hadoop-hdds/client/src/test/java/org/apache/hadoop/hdds/scm/client/TestHddsClientUtils.java

```diff
@@ -225,8 +232,38 @@ public void testVerifyResourceName() {
 
     for (String name : invalidNames) {
       assertThrows(IllegalArgumentException.class, () -> HddsClientUtils.verifyResourceName(name),
-          "Did not reject invalid string [" + name + "] as a name");
+          getInvalidNameMessage(name));
     }
+
+    // Message should include the name and resource type, if the name is of
+    // illegal length.
+    List<String> resourceTypes = ImmutableList.of("bucket", "volume");
+    for (int i = 0; i < 2; i++) {
+      String resType = resourceTypes.get(i);
+      String otherResType = resourceTypes.get(1 - i);
+      Throwable thrown = assertThrows(
+          IllegalArgumentException.class,
+          () -> HddsClientUtils.verifyResourceName(tooShort, resType, true),
+          getInvalidNameMessage(tooShort));
+
+      String message = thrown.getMessage();
+      assertNotNull(message);
+      assertTrue(message.contains(resType));
+      assertFalse(message.contains(otherResType));
+      assertTrue(thrown.getMessage().contains(tooShort));
```

Comment:
Please use `assertThat` instead of `assertTrue`/`assertFalse` for better failure message.  (See HDDS-9951 for details.)

Reply:
Sounds like a significant improvement indeed. Thank you for including the ticket link too. Updated.